src/quic_solana.rs

```diff
@@ -47,10 +47,21 @@ pub struct ConnectionCacheIdentity {
     shared: Arc<Mutex<QuicSessionInner>>,
     identity_watcher: watch::Sender<Pubkey>,
     reactive_signer: watch::Sender<PubkeySigner>,
+    stop_transactions: watch::Sender<()>,
+    allow_reset: watch::Receiver<()>,
 }
 
 impl ConnectionCacheIdentity {
-    pub async fn update_keypair(&self, keypair: &Keypair) {
+    pub async fn update_keypair(&self, keypair: &Keypair, is_reset: bool) {
+        let kp = keypair.insecure_clone();
+        let ps = PubkeySigner::new(kp);
+        self.reactive_signer.send_replace(ps);
+        if is_reset {
+            let mut allow_reset = self.allow_reset.clone();
+            self.stop_transactions.send_replace(());
```

Comment:
I think we should stop transactions regardless of whether the identity is a reset or not? if i nterpret transaction stopping in the right way